x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/create_engine_flyout.tsx

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFieldText,
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiLink,
+  EuiSpacer,
+  EuiSteps,
+  EuiText,
+  EuiTitle,
+  EuiComboBoxOptionOption,
+  EuiCallOut,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { Status } from '../../../../../common/types/api';
+import { ElasticsearchIndexWithIngestion } from '../../../../../common/types/indices';
+
+import { CANCEL_BUTTON_LABEL } from '../../../shared/constants';
+
+import { getErrorsFromHttpResponse } from '../../../shared/flash_messages/handle_api_errors';
+
+import { indexToOption, IndicesSelectComboBox } from './components/indices_select_combobox';
+
+import { CreateEngineLogic } from './create_engine_logic';
+
+export interface CreateEngineFlyoutProps {
+  onClose: () => void;
+}
+
+export const CreateEngineFlyout = ({ onClose }: CreateEngineFlyoutProps) => {
+  const { createEngine, setEngineName, setSelectedIndices } = useActions(CreateEngineLogic);
+  const {
+    createDisabled,
+    createEngineError,
+    createEngineStatus,
+    engineName,
+    engineNameStatus,
+    formDisabled,
+    indicesStatus,
+    selectedIndices,
+  } = useValues(CreateEngineLogic);
+
+  const onIndicesChange = (
+    selectedOptions: Array<EuiComboBoxOptionOption<ElasticsearchIndexWithIngestion>>
+  ) => {
+    setSelectedIndices(
+      selectedOptions
+        .map((option) => option.value)
+        .filter(
```

Comment:
nitpick: we have an `isNotNullish` function to achieve the same result in a generic way: https://github.com/elastic/kibana/tree/main/x-pack/plugins/enterprise_search/common/utils/is_not_nullish.ts